bot/cogs/moderation/utils.py

```diff
@@ -31,38 +31,45 @@
 Expiry = t.Union[Duration, ISODateTime]
 
 
-async def post_user(ctx: Context, user: discord.User) -> t.Optional[dict]:
+async def post_user(ctx: Context, user: t.Union[discord.User, discord.Object]) -> t.Optional[dict]:
     """
     Create a new user in the database.
 
     Used when an infraction needs to be applied on a user absent in the guild.
     """
-    log.warn("Attempting to add user to the database.")
+    log.trace("Attempting to add user to the database.")
 
-    payload = {
-        'avatar_hash': user.avatar,
-        'discriminator': int(user.discriminator),
-        'id': user.id,
-        'in_guild': False,
-        'name': user.name,
-        'roles': []
-    }
+    try:
+        payload = {
+            'avatar_hash': user.avatar,
+            'discriminator': int(user.discriminator),
+            'id': user.id,
+            'in_guild': False,
+            'name': user.name,
+            'roles': []
+        }
+    except AttributeError:
+        log.trace("Couldn't take all the attributes for the user payload, taking just its ID.")
+        # XXX: Not sure if these default values are ideal.
+        payload = {
+            'avatar_hash': 0,
+            'discriminator': 0,
+            'id': user.id,
+            'in_guild': False,
+            'name': 'Some name',
+            'roles': []
+        }
```

Comment:
My lack of experience doesn't tell me if doing `t.Union[discord.User, discord.Object]` and an indistinctive try-except is good practice or not. Need feeback on this.

Reply:
I just remembered `getattr` exists.